src/core/lib/security/credentials/google_default/google_default_credentials.cc

```diff
@@ -83,6 +84,21 @@ struct metadata_server_detector {
   int success;
   grpc_http_response response;
 };
+
+namespace {
+
+bool IsXdsNonCfeCluster(const char* xds_cluster) {
+  if (xds_cluster == nullptr) return false;
+  if (absl::StartsWith(xds_cluster, "google_cfe_")) return false;
+  if (!absl::StartsWith(xds_cluster, "xdstp:")) return true;
+  auto uri = grpc_core::URI::Parse(xds_cluster);
+  if (!uri.ok()) return true;  // Shouldn't happen, but assume ALTS.
```

Comment:
Since we don't expect this case to happen, can we log an error showing the cluster name and the uri parse error?

Reply:
I don't think that log message would ever actually be triggered.  If the cluster name starts with "xdstp:" but is not parseable as a URI, the XdsClient will report the failure to the watcher without ever subscribing to the resource on the wire, so we'll never actually get any endpoints with this kind of cluster name.